### PR TITLE
Added access control enforcer for every api handler

### PR DESCRIPTION
**Pull request by powerfooI**
## Summary

1. Added access control enforcer for every api handler
2. Fixed unexpected situation where storages are display in format of bytes #489
3. Fixed some problems in manifests

### CODE REVIEW

File: internal/dashboard/router/v1/obcluster_router.go [hunk 1]
```diff
@@ -15,19 +15,20 @@ package v1
...
-	g.GET("/obclusters/namespace/:namespace/name/:name", h.Wrap(h.GetOBCluster))
-	g.POST("/obclusters/namespace/:namespace/name/:name", h.Wrap(h.UpgradeOBCluster))
-	g.DELETE("/obclusters/namespace/:namespace/name/:name", h.Wrap(h.DeleteOBCluster))
-	g.POST("/obclusters/namespace/:namespace/name/:name/obzones", h.Wrap(h.AddOBZone))
-	g.POST("/obclusters/namespace/:namespace/name/:name/obzones/:obzoneName/scale", h.Wrap(h.ScaleOBServer))
-	g.DELETE("/obclusters/namespace/:namespace/name/:name/obzones/:obzoneName", h.Wrap(h.DeleteOBZone))
-	g.GET("/obclusters/:namespace/:name/resource-usages", h.Wrap(h.ListOBClusterResources))
-	g.GET("/obclusters/:namespace/:name/related-events", h.Wrap(h.ListOBClusterRelatedEvents))
+	g.GET("/obclusters/statistic", h.Wrap(h.GetOBClusterStatistic, acbiz.PathGuard("obcluster", "*", "read")))
+	g.GET("/obclusters", h.Wrap(h.ListOBClusters, acbiz.PathGuard("obcluster", "", "read")))
```
Comment (chris-sun-star) [inline internal/dashboard/router/v1/obcluster_router.go:24]: what's the difference between * and ""

File: internal/dashboard/business/ac/enforcers.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	"github.com/gin-contrib/sessions"
+	"github.com/gin-gonic/gin"
+)
+
+type EnforceFunc func(c *gin.Context) (bool, error)
+
+// groups of domain, resource and action
+// if resource is prefixed with ":", it means a parameter in the path
+// if resource contains "+", it means a list of resources, like :namespace+:name
+func PathGuard(perms ...string) EnforceFunc {
```
Comment (chris-sun-star) [inline internal/dashboard/business/ac/enforcers.go:27]: If you need to receive more than one resource permissions, pass them in a structured way

Comment (chris-sun-star) [reply internal/dashboard/business/ac/enforcers.go:27]: maybe there's no need to support mulltiple resource permissions

File: internal/dashboard/handler/wrapper.go [hunk 3]
```diff
@@ -34,8 +35,30 @@ func logHandlerError(c *gin.Context, err error) {
 		Error("handler error")
 }
 
-func Wrap[T any](h Handler[T]) gin.HandlerFunc {
+func Wrap[T any](h Handler[T], funcs ...acbiz.EnforceFunc) gin.HandlerFunc {
```
Comment (chris-sun-star) [inline internal/dashboard/handler/wrapper.go:38]: what's the main purpose to support passing multiple enforce functions, is it meant to describe a more strict permission by combining multiple expressions or a more loose policy if any condition fulfilled

File: internal/dashboard/business/oceanbase/obcluster.go [hunk 4]
```diff
@@ -127,7 +127,7 @@ func buildOBClusterResponse(ctx context.Context, obcluster *v1alpha1.OBCluster)
 		respCluster.Monitor.Image = obcluster.Spec.MonitorTemplate.Image
 		respCluster.Monitor.Resource = response.ResourceSpecRender{
 			Cpu:      obcluster.Spec.MonitorTemplate.Resource.Cpu.Value(),
-			MemoryGB: obcluster.Spec.MonitorTemplate.Resource.Memory.String(),
+			MemoryGB: fmt.Sprintf("%d GB", obcluster.Spec.MonitorTemplate.Resource.Memory.Value()>>30),
```
Comment (chris-sun-star) [inline internal/dashboard/business/oceanbase/obcluster.go:130]: return the actual value and parse it in frontend

File: internal/dashboard/router/v1/obcluster_router.go (hunk 1 above)
Comment (powerfooI) [reply internal/dashboard/router/v1/obcluster_router.go:24]: `*` means all of resource while empty string means any one of resource

File: internal/dashboard/business/oceanbase/obcluster.go (hunk 4 above)
Comment (powerfooI) [reply internal/dashboard/business/oceanbase/obcluster.go:130]: I don't think it's necessary to return the original value here. The response is just of type `ResourceSpecRender` for rendering. And the frontend need to change its code if we return original value here.

File: internal/dashboard/handler/wrapper.go (hunk 3 above)
Comment (powerfooI) [reply internal/dashboard/handler/wrapper.go:38]: It's a more strict pattern. By now there is only one enforcer used in practice.

File: internal/dashboard/business/ac/enforcers.go (hunk 2 above)
Comment (powerfooI) [reply internal/dashboard/business/ac/enforcers.go:27]: It's painful to put structs in API router registration. :( By now there is only one pair of domain, object and action used in practice.

**Review by cheer-chen — APPROVED** [review]
LGTM

**Review by chris-sun-star — APPROVED** [review]
LGTM

**Pull request merged**